From roel.kluin@gmail.com  Fri Nov  6 14:49:16 2009
From: Roel Kluin <roel.kluin@gmail.com>
Date: Tue, 03 Nov 2009 23:48:58 +0100
Subject: Staging: dst: Fix parentheses
To: Greg Kroah-Hartman <gregkh@suse.de>, devel@driverdev.osuosl.org, Andrew Morton <akpm@linux-foundation.org>, zbr@ioremap.net
Message-ID: <4AF0B35A.8080400@gmail.com>


`|' has a higher precedence than `?' so since MSG_WAITALL is
defined 0x100, MSG_MORE was always written to the msg_flag.

Signed-off-by: Roel Kluin <roel.kluin@gmail.com>
Acked-by: Evgeniy Polyakov <zbr@ioremap.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/dst/state.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/dst/state.c
+++ b/drivers/staging/dst/state.c
@@ -121,7 +121,7 @@ int dst_data_send_header(struct socket *
 	msg.msg_namelen = 0;
 	msg.msg_control = NULL;
 	msg.msg_controllen = 0;
-	msg.msg_flags = MSG_WAITALL | (more)?MSG_MORE:0;
+	msg.msg_flags = MSG_WAITALL | (more ? MSG_MORE : 0);
 
 	err = kernel_sendmsg(sock, &msg, &iov, 1, iov.iov_len);
 	if (err != size) {
